test/functional/ftplugin/help_spec.lua

```diff
@@ -0,0 +1,34 @@
+local t = require('test.testutil')
+local n = require('test.functional.testnvim')()
+local Screen = require('test.functional.ui.screen')
+
+local eq = t.eq
+
+describe('gx works for tags', function()
+  local screen --- @type test.functional.ui.screen
+  local link_ns
+
+  before_each(function()
+    n.clear({ args = { '-u', 'NORC' } })
+    screen = Screen.new(64, 16)
+    n.command('help nvim.txt')
+    link_ns = n.api.nvim_create_namespace('nvim.help.urls')
+  end)
+
+  it('creates extmarks for optionlink, taglink and tag nodes', function()
+    t.neq(0, #n.api.nvim_buf_get_extmarks(0, link_ns, 0, -1, {}))
+  end)
+
+  it('points to https://neovim.io/doc', function()
+    local link = n.api.nvim_buf_get_extmarks(
+      0, link_ns, 0, -1, { limit = 1, details = true }
+    )[1]
+
+    local url = link[4].url
+    eq(true, vim.startswith(url, 'https://neovim.io/doc'))
+
+    local param = url:match("%?tag=(.*)")
+    local tag = n.api.nvim_buf_get_text(0, link[2],link[3], link[4].end_row, link[4].end_col, {})[1]
+    eq(vim.uri_encode(tag), param)
+  end)
```

Comment:
Should I add a test that _tests_ a URL to see if it's valid using `vim.net.request`?